File: src(dossier java)/main/java/Ex/dto/AuthResponse.java
package Ex.dto;

import java.util.Set;

public class AuthResponse {

    private String token;
    private Long id;
    private String email;
    private String nom;
    private String prenom;
    private Set<String> roles;

    public AuthResponse() {}

    public AuthResponse(String token, Long id, String email, String nom, String prenom, Set<String> roles) {
        this.token = token;
        this.id = id;
        this.email = email;
        this.nom = nom;
        this.prenom = prenom;
        this.roles = roles;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }
}

File: lib(dossier flutter)/models/auth_response.dart
import 'user.dart';

class AuthResponse {
  final String token;
  final int? id;
  final String email;
  final String nom;
  final String prenom;
  final Set<String> roles;

  AuthResponse({
    required this.token,
    this.id,
    required this.email,
    required this.nom,
    required this.prenom,
    required this.roles,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'],
      id: json['id'],
      email: json['email'],
      nom: json['nom'],
      prenom: json['prenom'],
      roles: Set<String>.from(json['roles'] ?? []),
    );
  }

  User toUser() {
    return User(
      id: id,
      email: email,
      nom: nom,
      prenom: prenom,
      roles: roles,
    );
  }
}
